keysign/send.py

```diff
@@ -92,12 +99,46 @@ def __init__(self, builder=None):
         self.notify = None
         self.internet_option = False
 
+        # Add drag and drop to the keys list widget
+        builder.connect_signals(self)
+        self.label = builder.get_object("keys_listbox")
+        self.label.drag_dest_set(Gtk.DestDefaults.ALL, [], DRAG_ACTION)
+        self.label.drag_dest_set_target_list(None)
+        self.label.drag_dest_add_text_targets()
+        self.label.drag_dest_add_uri_targets()
+
+    def on_drag_data_received(self, widget, drag_context, x, y, data, info, time):
+        if self.notify is not None:
+            log.debug("We are trying to send a key, no imports at this stage")
+            return
+
+        filename = data.get_text()
+        # If we don't have a filename it means that the user maybe dropped
+        # an attachment or an entire email.
+        if not filename:
+            filename = data.get_data().decode("utf-8")
+        filename = filename[7:].strip('\r\n\x00')  # remove file://, \r\n and NULL
```

Comment:
Interesting. Is the filename URL encoded? That is, do we need to urldecode the filename? How is a character like "&" handled here?

Reply:
"&" was correctly shown. But there were problems with other characters like "%". I added unquote() to fix it.